### PR TITLE
Clean up initial proposal for first master merge

### PR DESCRIPTION
**Pull request by ybur-yug**
This includes:
- addition of static analysis via credo
- addition of new moduledocs where they were lacking
- removal of arbitrary test
- clean up credo warnings

### CODE REVIEW

File: lib/diplomat/plug/ensure_permitted.ex [hunk 1]
```diff
@@ -29,18 +30,13 @@ defmodule Diplomat.Plug.EnsurePermitted do
 
   defp fetch_resource({nil, conn}, opts), do: handle_error(:unauthenticated, conn, opts)
   defp fetch_resource({actor, conn}, opts) do
-    resource = Diplomat.Plug.current_resource(conn)
+    resource = Plug.current_resource(conn)
     {actor, resource, conn, opts}
   end
 
+  defp permitted?(conn, _), do: conn
```
Comment (doomspork) [inline lib/diplomat/plug/ensure_permitted.ex]: This has to go last doesn't it?

File: lib/diplomat/plug/load_resource.ex [hunk 2]
```diff
@@ -22,7 +23,7 @@ defmodule Diplomat.Plug.LoadResource do
   defp fetch_resource({loader, opts}, conn) do
     case apply(loader, :fetch_resource, [conn_tuple(conn), conn.params]) do
       {:ok, resource} ->
-        Plug.Conn.put_private(conn, :diplomat_resource, resource)
+        Conn.put_private(conn, :diplomat_resource, resource)
```
Comment (doomspork) [inline lib/diplomat/plug/load_resource.ex]: If this is the only thing we're using from `Plug.Conn` we could probably do: 

```elixir
import Plug.Conn, only: [put_private: 3]

... 
put_private(conn, :diplomat_resource, resource)
```

File: lib/diplomat/plug/ensure_permitted.ex (hunk 1 above)
Comment (ybur-yug) [reply lib/diplomat/plug/ensure_permitted.ex]: I think you're right. It didn't break tests so I didn't catch this. Just was a bit more readable to my eyes when I moved it.

Comment (doomspork) [reply lib/diplomat/plug/ensure_permitted.ex]: Then we need better tests!

**Pull request merged**